test/integration/oauth2_test.rb

```diff
@@ -65,13 +66,57 @@ def test_oauth2_pkce_s256
     test_token(token, user, client)
   end
 
+  def test_openid_connect
+    user = create(:user)
+    client = create(:oauth_application, :redirect_uri => "https://some.web.app.example.org/callback", :scopes => "openid read_prefs")
+    state = SecureRandom.urlsafe_base64(16)
+    verifier = SecureRandom.urlsafe_base64(48)
+    challenge = Base64.urlsafe_encode64(Digest::SHA256.digest(verifier), :padding => false)
+
+    authorize_client(user, client, :state => state, :code_challenge => challenge, :code_challenge_method => "S256", :scope => "openid read_prefs")
+    assert_response :redirect
+    code = validate_redirect(client, state)
+
+    tokens = request_tokens(client, code, verifier)
+    id_token = tokens[:id_token]
+    access_token = tokens[:access_token]
+
+    assert_not_nil(id_token)
+
+    data, _headers = JWT.decode id_token, Doorkeeper::OpenidConnect.signing_key.keypair, true, {
+      :algorithm => [Doorkeeper::OpenidConnect.signing_algorithm.to_s],
```

Comment:
We should probably get the key and algorithm the same way a real client will, by fetching `/oauth2/discovery/keys`.

In fact we should probably fetch `/.well-known/openid-configuration` and do some validation on it, then follow the `jwks_uri` entry to get the keys.

Reply:
This was my original intention. The problem is that in test environment discovery endpoint returns a mix of test host names in the remaining endpoints:
- `issuer` is based on server_url and server_protocol, consistently in test and development
- the remaining endpoints have correct paths, but have different hosts in development and test:
   - host in development is `http://localhost:3000` for all remaining endpoints
   - host in testing is `http://www.example.com`

For example, `jwks_uri` is set to `http://www.example.com/oauth2/discovery/keys`.
I am not too sure where the example.com comes from, and how it disapears in development...

I added a separate test for discovery endpoint which verifies that the endpoints have correct path. More importantly it gets the key informrmation from jwks endpoint (using path only) and verifies that kid is in fact kid of the configured key